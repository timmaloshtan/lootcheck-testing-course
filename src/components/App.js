import React, { Component } from 'react';

import Wallet from './Wallet';

export class App extends Component {
  componentDidMount() {
    fetch('https://api.coindesk.com/v1/bpi/currentprice.json')
      .then(res => res.json())
      .then(console.log);
  }
  
  render() {
    return (
      <div className="app">
        <h2>Loot Check</h2>
        <hr />
        <Wallet />
      </div>
    )
  }
}

export default App;
