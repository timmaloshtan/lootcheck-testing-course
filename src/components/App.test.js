import React from 'react';
import { shallow } from 'enzyme';

import App from './App';

describe('App component', () => {
  const app = shallow(<App />);

  it('should render correctly', () => {
    expect(app).toMatchSnapshot();
  });

  it('should contain a connected Wallet component', () => {
    expect(app.find('Connect(Wallet)').exists()).toBe(true);
  });
  
});
