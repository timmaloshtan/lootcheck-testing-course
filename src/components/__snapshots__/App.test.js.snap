// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
<div
  className="app"
>
  <h2>
    Loot Check
  </h2>
</div>
`;
